### PR TITLE
ci: Add release workflow for Sentry Publish

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Add release workflow script for releasing with the Sentry Central Publish Repository.

## :bulb: Motivation and Context

We want to unify how we do releases at Sentry.

## :green_heart: How did you test it?
Didn't. We will see if it works, once we try our first release. If it doesn't work out, we can always use craft locally.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I updated the CHANGELOG if needed
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -74,6 +74,8 @@ To make a header public follow these steps:
 * Add it to the Umbrella Header [Sentry.h](/Sources/Sentry/Public/Sentry.h).
 * Set the target membership to public.
 
+## Releasing
+For releasing a new version, see https://github.com/getsentry/publish#usage. The link is private to Sentry team members.
```
Comment (bruno-garcia) [inline CONTRIBUTING.md:78]: This document IMHO targets outside contributors I wouldn't add this since that's a private repo.

Comment (philipphofmann) [reply CONTRIBUTING.md]: I removed it and added it to our internal Notion page about releases.

**Review by marandaneto — APPROVED** [review]
LGTM.
would be possible to test it out of off this branch? I'd prefer to merge only if its tested otherwise we'd have a broken CI on master till getting a fix that might not be just a one liner

**Comment by philipphofmann** [comment]
> would be possible to test it out of off this branch? I'd prefer to merge only if its tested otherwise we'd have a broken CI on master till getting a fix that might not be just a one liner

It's not really possible to test it from this branch here. We basically just call the `getsentry/action-prepare-release@v1` Github Action. So if something doesn't work I think it's going to be a fix in this action. We still can use craft for doing releases and you have to trigger this action manually. So I think if it is broken the harm is not big and we can open a PR for a fix easily.

**Pull request merged**